drivers/sensors/sht4x.c

```diff
@@ -0,0 +1,827 @@
+      err = -EINVAL;
+      break;
+    }
+
+  nxmutex_unlock(&priv->devlock);
+  return err;
+}
+
+/****************************************************************************
+ * Name: sht4x_unlink
+ ****************************************************************************/
+
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+static int sht4x_unlink(FAR struct inode *inode)
+{
+  FAR struct sht4x_dev_s *priv;
+  int err;
+
+  DEBUGASSERT(inode->i_private != NULL);
+  priv = inode->i_private;
+
+  err = nxmutex_lock(&priv->devlock);
+  if (err) return err;
+
+  /* Are there open references to the driver data structure? */
+
+  if (priv->crefs <= 0)
+    {
+      nxmutex_destroy(&priv->devlock);
+      kmm_free(priv);
+      return 0;
+    }
+
+  /* No. Just mark the driver as unlinked and free the resources when
+   * the last client closes their reference to the driver.
+   */
+
+  priv->unlinked = true;
+  nxmutex_unlock(&priv->devlock);
+  return OK;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: sht4x_register
+ *
+ * Description:
+ *   Register the SHT4X character device as 'devpath'
+ *
+ * Input Parameters:
+ *   devpath - The full path to the driver to register. E.g., "/dev/temp0"
+ *   i2c     - An instance of the I2C interface to use to communicate with
+ *             the SHT4X
+ *   addr    - The I2C address of the SHT4X. The I2C address is one of 0x44,
+ *0x45 and 0x46.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int sht4x_register(FAR const char *devpath, FAR struct i2c_master_s *i2c,
+                   uint8_t addr)
+{
+  FAR struct sht4x_dev_s *priv;
+  int err;
+
+  DEBUGASSERT(i2c != NULL);
+  DEBUGASSERT(addr == 0x44 || addr == 0x45 || addr == 0x46);
+
+  /* Initialize the device structure */
+
+  priv = kmm_zalloc(sizeof(struct sht4x_dev_s));
+  if (priv == NULL)
+    {
+      snerr("ERROR: Failed to allocate instance.\n");
+      return -ENOMEM;
+    }
+
+  priv->i2c = i2c;
+  priv->addr = addr;
+  priv->precision = SHT4X_PREC_HIGH;
+  err = clock_systime_timespec(&priv->last_heat);
+
+  if (err)
+    {
+      snerr("ERROR: Failed to get timespec: %d\n", err);
+      kmm_free(priv);
+      return err;
+    }
+
+  /* Allow heat immediately after registration since in theory the sensor has
+   * never had its heater activated yet.
+   */
+
+  priv->last_heat.tv_sec -= 1;
+
+  err = nxmutex_init(&priv->devlock);
+  if (err)
+    {
+      snerr("ERROR: Failed to register SHT4X driver: %d\n", err);
+      kmm_free(priv);
+      return err;
+    }
+
+  /* Register the character driver */
+
+  err = register_driver(devpath, &g_sht4xfops, 0666, priv);
+  if (err < 0)
```

Comment:
let's unify the error ccheck by `if (err < 0)` instead `if (err)`

Reply:
Do you know why the check is typically explicitly for negatives?